src/types/serialization.js

```diff
@@ -50,12 +50,12 @@ export function serialize(buffer, shift, data, type) {
                 buffer = serialize(buffer, from, fieldData, fieldType.type);
             } else {
                 var end = buffer.length;
-                Uint32(end, buffer, from, from + 4);
+                Uint32(end - shift, buffer, from, from + 4);
                 buffer = serialize(buffer, end, fieldData, fieldType.type);
                 Uint32(buffer.length - end, buffer, from + 4, from + 8);
             }
         } else {
-            buffer = fieldType.type(fieldData, buffer, from, shift + fieldType.to);
+            buffer = fieldType.type(fieldData, buffer, from, shift + fieldType.to, buffer.length - shift);
```

Comment:
What values can `fieldType.type` assume here? Only `String`?

Reply:
1. `String`
2. `FixedBuffer`
3. entity of `Exonum.newType`